samples_skills/conversationalRetrievalwithLangchain.ipynb

```diff
@@ -0,0 +1,266 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this notebook we will demonstrate how to use LangChain to perform [conversational retrieval](https://python.langchain.com/docs/use_cases/question_answering/chat_vector_db).\n",
+    "\n",
+    "In a conversational question and answering scenario, users often pose follow-up questions related to the same topic, with the context being crucial to understand their queries. To address such cases effectively, we use the ConversationalRetrievalChain. Behind the scenes, this chain takes the user's question and converts it into a standalone query by considering the conversation history. Subsequently, it uses this standalone question to query the search service for relevant information."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! pip install azure-identity"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this notebook we are utilizing an existing search index we already set up with Azure Cognitive Search. You can follow this [link](https://python.langchain.com/docs/integrations/vectorstores/azuresearch) to create your own search index. In our search index, the content vector field is named \"contentVector\", so we set it as an environment variable below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "os.environ[\"AZURESEARCH_FIELDS_CONTENT_VECTOR\"] = \"contentVector\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain.embeddings.openai import OpenAIEmbeddings\n",
+    "from langchain.llms import AzureOpenAI\n",
+    "from langchain.chains import ConversationalRetrievalChain\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "True"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from dotenv import load_dotenv\n",
+    "load_dotenv(\"../llm.env\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain.vectorstores.azuresearch import AzureSearch\n",
```

Comment:
What version of langchain are you  using? i'm using 0.0.267 and i get the following error:

```
ImportError                               Traceback (most recent call last)
[c:\Users\JOMCDOWE\OneDrive](file:///C:/Users/JOMCDOWE/OneDrive) - Microsoft\Documents\Trident\rag-e2e-samples\samples_skills\conversationalRetrievalwithLangchain.ipynb Cell 7 line 9
      [7](vscode-notebook-cell:/c%3A/Users/JOMCDOWE/OneDrive%20-%20Microsoft/Documents/Trident/rag-e2e-samples/samples_skills/conversationalRetrievalwithLangchain.ipynb#W6sZmlsZQ%3D%3D?line=6) vector_store_address = os.getenv("COGSEARCH_ADDRESS")
      [8](vscode-notebook-cell:/c%3A/Users/JOMCDOWE/OneDrive%20-%20Microsoft/Documents/Trident/rag-e2e-samples/samples_skills/conversationalRetrievalwithLangchain.ipynb#W6sZmlsZQ%3D%3D?line=7) vector_store_password = os.getenv("COGSEARCH_API_KEY")
----> [9](vscode-notebook-cell:/c%3A/Users/JOMCDOWE/OneDrive%20-%20Microsoft/Documents/Trident/rag-e2e-samples/samples_skills/conversationalRetrievalwithLangchain.ipynb#W6sZmlsZQ%3D%3D?line=8) vector_store = AzureSearch(
     [10](vscode-notebook-cell:/c%3A/Users/JOMCDOWE/OneDrive%20-%20Microsoft/Documents/Trident/rag-e2e-samples/samples_skills/conversationalRetrievalwithLangchain.ipynb#W6sZmlsZQ%3D%3D?line=9)     azure_search_endpoint=vector_store_address,
     [11](vscode-notebook-cell:/c%3A/Users/JOMCDOWE/OneDrive%20-%20Microsoft/Documents/Trident/rag-e2e-samples/samples_skills/conversationalRetrievalwithLangchain.ipynb#W6sZmlsZQ%3D%3D?line=10)     azure_search_key=vector_store_password,
     [12](vscode-notebook-cell:/c%3A/Users/JOMCDOWE/OneDrive%20-%20Microsoft/Documents/Trident/rag-e2e-samples/samples_skills/conversationalRetrievalwithLangchain.ipynb#W6sZmlsZQ%3D%3D?line=11)     index_name=index_name,
     [13](vscode-notebook-cell:/c%3A/Users/JOMCDOWE/OneDrive%20-%20Microsoft/Documents/Trident/rag-e2e-samples/samples_skills/conversationalRetrievalwithLangchain.ipynb#W6sZmlsZQ%3D%3D?line=12)     embedding_function=embeddings.embed_query,
     [14](vscode-notebook-cell:/c%3A/Users/JOMCDOWE/OneDrive%20-%20Microsoft/Documents/Trident/rag-e2e-samples/samples_skills/conversationalRetrievalwithLangchain.ipynb#W6sZmlsZQ%3D%3D?line=13) )

File [c:\Users\JOMCDOWE\Miniconda3\envs\nbdev\lib\site-packages\langchain\vectorstores\azuresearch.py:231](file:///C:/Users/JOMCDOWE/Miniconda3/envs/nbdev/lib/site-packages/langchain/vectorstores/azuresearch.py:231), in AzureSearch.__init__(self, azure_search_endpoint, azure_search_key, index_name, embedding_function, search_type, semantic_configuration_name, semantic_query_language, fields, vector_search, semantic_settings, scoring_profiles, default_scoring_profile, **kwargs)
    207 self.embedding_function = embedding_function
    208 default_fields = [
    209     SimpleField(
    210         name=FIELDS_ID,
   (...)
    229     ),
    230 ]
--> 231 self.client = _get_search_client(
    232     azure_search_endpoint,
    233     azure_search_key,
    234     index_name,
    235     semantic_configuration_name=semantic_configuration_name,
...
     91 )
     93 default_fields = default_fields or []
     94 if key is None:

ImportError: cannot import name 'VectorSearchAlgorithmConfiguration' from 'azure.search.documents.indexes.models' (c:\Users\JOMCDOWE\Miniconda3\envs\nbdev\lib\site-packages\azure\search\documents\indexes\models\__init__.py)
```

Reply:
the latest version doesn't work for me either

Reply:
this is what I have: 0.0.327
I see we didn't impose a strict version number in the env yaml